python_scripts/01_tabular_data_exploration.py

```diff
@@ -364,5 +364,5 @@
 #   and evaluating your machine learning model;
 # * having redundant (or highly correlated) columns can be a problem for some
 #   machine learning algorithms;
-# * contrary to decision tree, linear models can only capture linear
-#   interactions, so be aware of non-linear relationships in your data.
+# * decision trees create rules at constant values of the features, such as
+#   vertical and horizontal lines in a 2-dimensional feature space.
```

Comment:
```suggestion
# * decision trees define prediction rules by comparing feature values to thresholds,
#   one feature after another. The resulting decision boundary of a decision tree
#   is therefore composed of vertical and horizontal line segments in a 2-dimensional
#   feature space.
```

Reply:
Thanks for the review @ogrisel. I addressed your other comments, but this wording feels a bit too complex for an introductory notebook.

Reply:
> Thanks for the review @ogrisel. I addressed your other comments, but this wording feels a bit too complex for an introductory notebook.

Personally, I find the previous wording more confusing: "create rules at constant values of the features" does not mean anything. What does a "rule at a value" mean?

Maybe the following is both simpler and actually meaningful:

```suggestion
# * decision trees define prediction rules that successively compare each feature to
#   a threshold value. The resulting decision boundary of a decision tree is
#   therefore composed of vertical and horizontal line segments in a 2-dimensional
#   feature space.
```